diff --git a/lang/c/examples/quickstop.c b/lang/c/examples/quickstop.c
index ff9e970..b26dad1 100644
--- a/lang/c/examples/quickstop.c
+++ b/lang/c/examples/quickstop.c
@@ -107,7 +107,7 @@ int print_person(avro_file_reader_t db, avro_schema_t reader_schema)
 	if (rval == 0) {
 		int64_t id;
 		int32_t age;
-		int32_t *p;
+		const char *p;
 		size_t size;
 		avro_value_t id_value;
 		avro_value_t first_value;
