--- installed/ksl/include/avx2ki_type.h	2024-05-28 16:04:15.099369984 +0800
+++ installed/ksl/include/avx2ki_type.h_1	2024-05-28 16:08:20.399369984 +0800
@@ -20,6 +20,7 @@ extern "C" {
 
 #pragma push_macro("FORCE_INLINE")
 #pragma push_macro("ALIGN_STRUCT")
+#undef FORCE_INLINE
 #define FORCE_INLINE static inline __attribute__((always_inline))
 #define ALIGN_STRUCT(x) __attribute__((aligned(x)))
 #define FUNC_HIDDEN __attribute__((visibility("hidden")))
@@ -141,7 +142,7 @@ __extension__
 #define _CMP_GT_OQ 0x1e
 #define _CMP_TRUE_US 0x1f
 
-#define GCC_VERSION (__GNUC__ * 100 + __GNUC_MINOR__ * 10 + __GNUC_PATCHLEVEL__)
+#define GCC_VERSION_2 (__GNUC__ * 100 + __GNUC_MINOR__ * 10 + __GNUC_PATCHLEVEL__)
 
 #if (!__clang__) && __GNUC__ >= 4 && __GNUC__ < 5
     typedef int16_t float16_t;
@@ -216,7 +217,7 @@ typedef union {
     __m64 vect_m64[2];
 } __m128d __attribute__((aligned(16)));
 
-#if defined(GCC_VERSION) && GCC_VERSION < 730
+#if defined(GCC_VERSION_2) && GCC_VERSION_2 < 730
 #else
 typedef union {
     float16_t vect_bf16[8];
@@ -264,7 +265,7 @@ typedef union {
     __m128d vect_f128[2];
 } __m256d __attribute__((aligned(32)));
 
-#if defined(GCC_VERSION) && GCC_VERSION < 730
+#if defined(GCC_VERSION_2) && GCC_VERSION_2 < 730
 #else
 typedef struct {
     float16x8_t vect_f16[2];
@@ -316,7 +317,7 @@ typedef union {
     __m256d vect_f256[2];
 } __m512d __attribute__((aligned(64)));
 
-#if defined(GCC_VERSION) && GCC_VERSION < 730
+#if defined(GCC_VERSION_2) && GCC_VERSION_2 < 730
 #else
 typedef struct {
     float16x8_t vect_f16[4];
