diff --git a/prefetch_crt_dependency.sh b/prefetch_crt_dependency.sh
index e7b2f94631e..54a179f2732 100755
--- a/prefetch_crt_dependency.sh
+++ b/prefetch_crt_dependency.sh
@@ -26,7 +26,7 @@ rm -rf crt
 echo "Downloading and expanding CRT CPP"
 mkdir -p crt/tmp
 curl ${CRT_URI} --output crt/tmp/crt.zip
-unzip crt/tmp/crt -d crt
+unzip -q crt/tmp/crt -d crt
 CRT_DIR_TMP=`ls -d crt/aws-crt-cpp*`
 mv ${CRT_DIR_TMP} crt/aws-crt-cpp
 
@@ -47,7 +47,7 @@ curl ${S2N_URI} --output crt/tmp/13.zip
 
 
 for a in `seq 1 13`; do
-    unzip crt/tmp/${a} -d crt/tmp
+    unzip -q crt/tmp/${a} -d crt/tmp
 done
 
 rm -rf crt/aws-crt-cpp/crt
