--- crt/aws-crt-cpp/crt/s2n/CMakeLists.txt.orig	2022-01-22 12:00:14.626738952 +0800
+++ crt/aws-crt-cpp/crt/s2n/CMakeLists.txt	2022-01-22 12:00:24.564666399 +0800
@@ -182,6 +182,7 @@
             "${CMAKE_CURRENT_LIST_DIR}/tests/features/noop_main.c"
             COMPILE_DEFINITIONS ${BIKE_R3_VPCLMUL_FLAGS})
 
+		set(BIKE_R3_VPCLMUL_SUPPORTED OFF)
         if(BIKE_R3_AVX2_SUPPORTED OR BIKE_R3_AVX512_SUPPORTED OR BIKE_R3_PCLMUL_SUPPORTED OR BIKE_R3_VPCLMUL_SUPPORTED)
             set(BIKE_R3_X86_64_OPT_SUPPORTED ON)
         endif()
