--- boost_1_75_0/tools/build/src/engine/build.sh	2020-12-03 13:02:49.000000000 +0800
+++ ../boost_1_75_0/tools/build/src/engine/build.sh	2020-12-14 18:44:20.327112548 +0800
@@ -232,7 +232,7 @@ case $B2_TOOLSET in
         ;;
 
         *)
-        B2_CXX="${CXX} -x c++ -std=c++11"
+        B2_CXX="${CXX} -x c++ -std=c++11  -static-libstdc++ -static-libgcc"
         B2_CXXFLAGS_RELEASE="-O2 -s"
         B2_CXXFLAGS_DEBUG="-O0 -g"
         esac
