diff -Naur cctz-2.4-step1/CMakeLists.txt cctz-2.4-final/CMakeLists.txt
--- cctz-2.4-step1/CMakeLists.txt	2026-03-26 10:51:44.734670986 +0000
+++ cctz-2.4-final/CMakeLists.txt	2026-03-26 10:50:03.612066877 +0000
@@ -196,3 +196,5 @@
 endif()
 
 feature_summary(WHAT ALL FATAL_ON_MISSING_REQUIRED_PACKAGES ${quiet_on_empty})
+
+SET(CMAKE_CXX_FLAGS "-fno-omit-frame-pointer -O3 -g -gdwarf-4  ${CMAKE_CXX_FLAGS}")
diff -Naur cctz-2.4-step1/include/cctz/civil_time_detail.h cctz-2.4-final/include/cctz/civil_time_detail.h
--- cctz-2.4-step1/include/cctz/civil_time_detail.h	2026-03-26 10:51:44.735702301 +0000
+++ cctz-2.4-final/include/cctz/civil_time_detail.h	2026-03-26 10:50:03.611905816 +0000
@@ -200,13 +200,13 @@
 CONSTEXPR_F fields n_sec(year_t y, diff_t m, diff_t d, diff_t hh, diff_t mm,
                          diff_t ss) noexcept {
   // Optimization for when (non-constexpr) fields are already normalized.
-  if (0 <= ss && ss < 60) {
+  if (__builtin_expect(0 <= ss && ss < 60, 1)) {
     const second_t nss = static_cast<second_t>(ss);
-    if (0 <= mm && mm < 60) {
+    if (__builtin_expect(0 <= mm && mm < 60, 1)) {
       const minute_t nmm = static_cast<minute_t>(mm);
-      if (0 <= hh && hh < 24) {
+      if (__builtin_expect(0 <= hh && hh < 24, 1)) {
         const hour_t nhh = static_cast<hour_t>(hh);
-        if (1 <= d && d <= 28 && 1 <= m && m <= 12) {
+        if (__builtin_expect(1 <= d && d <= 28 && 1 <= m && m <= 12, 1)) {
           const day_t nd = static_cast<day_t>(d);
           const month_t nm = static_cast<month_t>(m);
           return fields(y, nm, nd, nhh, nmm, nss);
diff -Naur cctz-2.4-step1/include/cctz/time_zone.h cctz-2.4-final/include/cctz/time_zone.h
--- cctz-2.4-step1/include/cctz/time_zone.h	2026-03-26 10:51:44.735758716 +0000
+++ cctz-2.4-final/include/cctz/time_zone.h	2026-03-26 10:50:03.611518573 +0000
@@ -95,6 +95,14 @@
     return lookup(detail::split_seconds(tp).first);
   }
 
+  // Lookup the UTC offset for a timepoint
+  // The returned the lookup result doesn't contain valid cs
+  absolute_lookup lookup_offset(const time_point<seconds> &tp) const;
+  template <typename D>
+  absolute_lookup lookup_offset(const time_point<D> &tp) const {
+    return lookup_offset(detail::split_seconds(tp).first);
+  }
+
   // A civil_lookup represents the absolute time(s) (time_point) that
   // correspond to the given civil time (cctz::civil_second) within this
   // time_zone. Usually the given civil time represents a unique instant
diff -Naur cctz-2.4-step1/src/cctz_benchmark.cc cctz-2.4-final/src/cctz_benchmark.cc
--- cctz-2.4-step1/src/cctz_benchmark.cc	2026-03-26 10:51:44.776493135 +0000
+++ cctz-2.4-final/src/cctz_benchmark.cc	2026-03-26 10:50:18.447282072 +0000
@@ -754,6 +754,24 @@
 }
 BENCHMARK(BM_Time_ToCivil_Cached);
 
+void BM_Time_ToCivil_LookupOffset_Batch1024(benchmark::State &state) {
+  const cctz::time_zone tz = TestTimeZone();
+  std::chrono::system_clock::time_point tp =
+      std::chrono::system_clock::from_time_t(1750656243);
+  std::chrono::system_clock::time_point tp2 =
+      std::chrono::system_clock::from_time_t(1750656245);
+  while (state.KeepRunning()) {
+    std::swap(tp, tp2);
+    // benchmark::DoNotOptimize(tz.lookup(tp).cs);
+
+    for (int i = 0; i < 1024; i++) {
+      tp += std::chrono::seconds(1);
+      benchmark::DoNotOptimize(tz.lookup_offset(tp).cs);
+    }
+  }
+}
+BENCHMARK(BM_Time_ToCivil_LookupOffset_Batch1024);
+
 void BM_Time_ToCivil_Libc(benchmark::State& state) {
   // No timezone support, so just use localtime.
   time_t t = 1384569027;
diff -Naur cctz-2.4-step1/src/time_zone_if.h cctz-2.4-final/src/time_zone_if.h
--- cctz-2.4-step1/src/time_zone_if.h	2026-03-26 10:51:44.736515659 +0000
+++ cctz-2.4-final/src/time_zone_if.h	2026-03-26 10:50:03.611032085 +0000
@@ -37,6 +37,10 @@
 
   virtual time_zone::absolute_lookup BreakTime(
       const time_point<seconds>& tp) const = 0;
+  virtual time_zone::absolute_lookup
+  LookupOffset(const time_point<seconds> &tp) const {
+    return BreakTime(tp);
+  }
   virtual time_zone::civil_lookup MakeTime(
       const civil_second& cs) const = 0;
 
diff -Naur cctz-2.4-step1/src/time_zone_impl.h cctz-2.4-final/src/time_zone_impl.h
--- cctz-2.4-step1/src/time_zone_impl.h	2026-03-26 10:51:44.736616578 +0000
+++ cctz-2.4-final/src/time_zone_impl.h	2026-03-26 10:50:03.611202162 +0000
@@ -50,6 +50,10 @@
     return zone_->BreakTime(tp);
   }
 
+  time_zone::absolute_lookup LookupOffset(const time_point<seconds> &tp) const {
+    return zone_->LookupOffset(tp);
+  }
+
   // Converts the civil-time components in this time zone into a time_point.
   // That is, the opposite of BreakTime(). The requested civil time may be
   // ambiguous or illegal due to a change of UTC offset.
diff -Naur cctz-2.4-step1/src/time_zone_info.cc cctz-2.4-final/src/time_zone_info.cc
--- cctz-2.4-step1/src/time_zone_info.cc	2026-03-26 10:51:44.776300566 +0000
+++ cctz-2.4-final/src/time_zone_info.cc	2026-03-26 10:49:28.751771964 +0000
@@ -884,6 +884,64 @@
   return (r < 0) ? q - 1 : q;
 }
 
+time_zone::absolute_lookup
+TimeZoneInfo::Offset(std::int_fast64_t unix_time,
+                     const TransitionType &tt) const {
+  return {{}, tt.utc_offset, tt.is_dst, &abbreviations_[tt.abbr_index]};
+}
+
+time_zone::absolute_lookup TimeZoneInfo::Offset(std::int_fast64_t unix_time,
+                                                const Transition &tr) const {
+  const TransitionType &tt = transition_types_[tr.type_index];
+  // std::int_fast64_t local_unix = unix_time + tt.utc_offset;
+  return {{}, tt.utc_offset, tt.is_dst, &abbreviations_[tt.abbr_index]};
+}
+
+time_zone::absolute_lookup
+TimeZoneInfo::LookupOffset(const time_point<seconds> &tp) const {
+  std::int_fast64_t unix_time = ToUnixSeconds(tp);
+  const std::size_t timecnt = transitions_.size();
+  assert(timecnt != 0); // We always add a transition.
+
+  const std::size_t hint = local_time_hint_.load(std::memory_order_relaxed);
+  // This branch is likely to be taken if the hint is valid and the unix_time
+  // falls within the hinted transition range.
+  if (0 < hint && hint < timecnt) [[likely]] {
+    if (transitions_[hint - 1].unix_time <= unix_time) [[likely]] {
+      if (unix_time < transitions_[hint].unix_time) [[likely]] {
+        return Offset(unix_time, transitions_[hint - 1]);
+      }
+    }
+  }
+
+  if ((unix_time < transitions_[0].unix_time)) {
+    return Offset(unix_time, transition_types_[default_transition_type_]);
+  }
+  if (unix_time >= transitions_.back().unix_time) {
+    // After the last transition. If we extended the transitions using
+    // future_spec_, shift back to a supported year using the 400-year
+    // cycle of calendaric equivalence and then compensate accordingly.
+    if (extended_) {
+      const std::int_fast64_t diff =
+          unix_time - transitions_[timecnt - 1].unix_time;
+      const year_t shift = diff / kSecsPer400Years + 1;
+      const auto d = seconds(shift * kSecsPer400Years);
+      time_zone::absolute_lookup al = BreakTime(tp - d);
+      al.cs = YearShift(al.cs, shift * 400);
+      return al;
+    }
+    return Offset(unix_time, transitions_[timecnt - 1]);
+  }
+
+  const Transition target = {unix_time, 0, civil_second(), civil_second()};
+  const Transition *begin = &transitions_[0];
+  const Transition *tr = std::upper_bound(begin, begin + timecnt, target,
+                                          Transition::ByUnixTime());
+  local_time_hint_.store(static_cast<std::size_t>(tr - begin),
+                         std::memory_order_relaxed);
+  return Offset(unix_time, *--tr);
+}
+
 // BreakTime() translation for a particular transition.
 time_zone::absolute_lookup TimeZoneInfo::LocalTime(std::int_fast64_t unix_time,
                                                    const Transition &tr) const {
diff -Naur cctz-2.4-step1/src/time_zone_info.h cctz-2.4-final/src/time_zone_info.h
--- cctz-2.4-step1/src/time_zone_info.h	2026-03-26 10:51:44.736764744 +0000
+++ cctz-2.4-final/src/time_zone_info.h	2026-03-26 10:50:03.611352463 +0000
@@ -68,6 +68,8 @@
   // TimeZoneIf implementations.
   time_zone::absolute_lookup BreakTime(
       const time_point<seconds>& tp) const override;
+  time_zone::absolute_lookup
+  LookupOffset(const time_point<seconds> &tp) const override;
   time_zone::civil_lookup MakeTime(
       const civil_second& cs) const override;
   bool NextTransition(const time_point<seconds>& tp,
@@ -93,8 +95,13 @@
   bool Load(ZoneInfoSource* zip);
 
   // Helpers for BreakTime() and MakeTime().
+  time_zone::absolute_lookup Offset(std::int_fast64_t unix_time,
+                                    const Transition &tr) const;
+  time_zone::absolute_lookup Offset(std::int_fast64_t unix_time,
+                                    const TransitionType &tt) const;
+
   time_zone::absolute_lookup LocalTime(std::int_fast64_t unix_time,
-                                       const TransitionType& tt) const;
+                                       const TransitionType &tt) const;
   time_zone::absolute_lookup LocalTime(std::int_fast64_t unix_time,
                                        const Transition& tr) const;
   time_zone::civil_lookup TimeLocal(const civil_second& cs,
diff -Naur cctz-2.4-step1/src/time_zone_lookup.cc cctz-2.4-final/src/time_zone_lookup.cc
--- cctz-2.4-step1/src/time_zone_lookup.cc	2026-03-26 10:51:44.736896480 +0000
+++ cctz-2.4-final/src/time_zone_lookup.cc	2026-03-26 10:50:03.611718867 +0000
@@ -173,6 +173,11 @@
   return effective_impl().BreakTime(tp);
 }
 
+time_zone::absolute_lookup
+time_zone::lookup_offset(const time_point<seconds> &tp) const {
+  return effective_impl().LookupOffset(tp);
+}
+
 time_zone::civil_lookup time_zone::lookup(const civil_second& cs) const {
   return effective_impl().MakeTime(cs);
 }
