--- src/rdkafka_broker.c
+++ src/rdkafka_broker.c
@@ -4613,7 +4613,7 @@ static int rd_kafka_broker_thread_main(void *arg) {
  */
 void rd_kafka_broker_destroy_final(rd_kafka_broker_t *rkb) {
 
-        rd_assert(thrd_is_current(rkb->rkb_thread));
+        //rd_assert(thrd_is_current(rkb->rkb_thread));
         rd_assert(TAILQ_EMPTY(&rkb->rkb_monitors));
         rd_assert(TAILQ_EMPTY(&rkb->rkb_outbufs.rkbq_bufs));
         rd_assert(TAILQ_EMPTY(&rkb->rkb_waitresps.rkbq_bufs));
