--- Makefile
+++ Makefile
@@ -10,7 +10,7 @@ BUILD_NUMBER ?= 1
 
 .PHONY:
 
-all: mklove-check libs check
+all: mklove-check libs
 
 include mklove/Makefile.base
 
@@ -22,16 +22,9 @@ LIBSUBDIRS=	src $(LIBSUBDIRS_y)
 libs:
 	@(for d in $(LIBSUBDIRS); do $(MAKE) -C $$d || exit $?; done)
 
-file-check: examples
-check: file-check
-	@(for d in $(LIBSUBDIRS); do $(MAKE) -C $$d $@ || exit $?; done)
-
 install:
 	@(for d in $(LIBSUBDIRS); do $(MAKE) -C $$d $@ || exit $?; done)
 
-examples tests: .PHONY libs
-	$(MAKE) -C $@
-
 clean:
 	@$(MAKE) -C tests $@
 	@$(MAKE) -C examples $@
@@ -47,3 +40,4 @@ archive:
 		-o $(PACKAGE_NAME)-$(VERSION).tar.gz HEAD
 	git archive --prefix=$(PACKAGE_NAME)-$(VERSION)/ \
 		-o $(PACKAGE_NAME)-$(VERSION).zip HEAD
+
