diff --git a/pulsar-client-cpp/lib/CompressionCodecZstd.cc b/pulsar-client-cpp/lib/CompressionCodecZstd.cc
index 14c09925501dafc6cb8d7540fa5c5c169da7baea..74e533f70fe87152e9c647f4088d1261316a5a18 100644
--- a/pulsar-client-cpp/lib/CompressionCodecZstd.cc
+++ b/pulsar-client-cpp/lib/CompressionCodecZstd.cc
@@ -19,7 +19,7 @@
 #include "CompressionCodecZstd.h"
 
 #if HAS_ZSTD
-#include <zstd.h>
+#include <zstd/zstd.h>
 
 namespace pulsar {
 
