diff --git a/db/blob/blob_file_meta.h b/db/blob/blob_file_meta.h
index 247f490..3a2a1e2 100644
--- a/db/blob/blob_file_meta.h
+++ b/db/blob/blob_file_meta.h
@@ -8,4 +8,5 @@
 #include <cassert>
+#include <cstdint>
 #include <iosfwd>
 #include <memory>
 #include <string>
diff --git a/include/rocksdb/sst_partitioner.h b/include/rocksdb/sst_partitioner.h
index 33e9062..080f0c6 100644
--- a/include/rocksdb/sst_partitioner.h
+++ b/include/rocksdb/sst_partitioner.h
@@ -8,2 +8,3 @@
 #include <memory>
+#include <cstdint>
 #include <string>
diff --git a/util/string_util.cc b/util/string_util.cc
index 6acb9de..8e6b6f5 100644
--- a/util/string_util.cc
+++ b/util/string_util.cc
@@ -121,3 +121,3 @@ std::string NumberToString(uint64_t num) {
 std::string NumberToHumanString(int64_t num) {
-  char buf[19];
+  char buf[32];
   int64_t absnum = num < 0 ? -num : num;
